sklearn/datasets/_samples_generator.py

```diff
@@ -523,6 +523,21 @@ def make_hastie_10_2(n_samples=12000, *, random_state=None):
     return X, y
 
 
+@validate_params(
+    {
+        "n_samples": [Interval(Integral, 1, None, closed="left")],
+        "n_features": [Interval(Integral, 1, None, closed="left")],
+        "n_informative": [Interval(Integral, 1, None, closed="left")],
```

Comment:
Let's make ``n_informative=0`` valid in case someone wants to generate pure noise
```suggestion
        "n_informative": [Interval(Integral, 0, None, closed="left")],
```